slasher/db/indexed_attestations.go

```diff
@@ -225,31 +207,29 @@ func (db *Store) SaveIndexedAttestation(idxAttestation *ethpb.IndexedAttestation
 	return err
 }
 
-func saveCompressedIdxAttToEpochList(idxAttestation *ethpb.IndexedAttestation, tx *bolt.Tx) error {
-	dataRoot, err := hashutil.HashProto(idxAttestation.Data)
-	if err != nil {
-		return errors.Wrap(err, "failed to hash indexed attestation data.")
-	}
-	protoIdxAtt := &slashpb.CompressedIdxAtt{
-		Signature: idxAttestation.Signature,
-		Indices:   idxAttestation.AttestingIndices,
-		DataRoot:  dataRoot[:],
-	}
-
-	key := bytesutil.Bytes8(idxAttestation.Data.Target.Epoch)
-	bucket := tx.Bucket(compressedIdxAttsBucket)
-	enc := bucket.Get(key)
-	compressedIdxAttList, err := unmarshalCompressedIdxAttList(enc)
-	if err != nil {
-		return errors.Wrap(err, "failed to decode value into CompressedIdxAtt")
-	}
-	compressedIdxAttList.List = append(compressedIdxAttList.List, protoIdxAtt)
-	enc, err = proto.Marshal(compressedIdxAttList)
-	if err != nil {
-		return errors.Wrap(err, "failed to marshal")
-	}
-	if err := bucket.Put(key, enc); err != nil {
-		return errors.Wrap(err, "failed to save indexed attestation into historical bucket")
+func saveEpochValidatorIdxAttList(idxAttestation *ethpb.IndexedAttestation, tx *bolt.Tx) error {
+	bucket := tx.Bucket(epochValidatorIdxAttsBucket)
+	idxAttKey := encodeEpochSig(idxAttestation.Data.Target.Epoch, idxAttestation.Signature)
+	for _, valIdx := range idxAttestation.AttestingIndices {
+		key := encodeEpochValidatorID(idxAttestation.Data.Target.Epoch, valIdx)
+		enc := bucket.Get(key)
+		if enc == nil {
+			if err := bucket.Put(key, idxAttKey); err != nil {
+				return errors.Wrap(err, "failed to save indexed attestation into historical bucket")
+			}
+		}
+		keys, err := unmarshalIdxAttKeys(enc)
```

Comment:
this loops through all keys for each attestation index. This will be very slow here

Reply:
+1

Reply:
@nisdas did you see unmarshalIdxAttKeys?

Reply:
should be only one key at most cases